enterprise/server/workflow/service/service.go

```diff
@@ -1162,14 +1160,37 @@ func (ws *workflowService) createActionForWorkflow(ctx context.Context, wf *tabl
 		}
 	}
 
+	// NOTE: For non-Linux workflows, the executor is responsible for ensuring the
+	// buildbuddy_ci_runner binary exists at the workspace root when it sees
+	// the `workflow-id` platform property.
+	inputRootDigest, err := digest.ComputeForMessage(&repb.Directory{}, repb.DigestFunction_SHA256)
+	if err != nil {
+		return nil, err
+	}
+	if os == "" || os == platform.LinuxOperatingSystemName {
+		// On Linux, because we can build the binary with the apps with the right architecture,
```

Comment:
nit: check arch as well, since we are starting to get more `linux/arm64` users

```suggestion
	if (os == "" || os == platform.LinuxOperatingSystemName) && (arch == "" || arch == "amd64") {
		// On Linux, because the apps are built for linux/amd64,
```

Reply:
+1